mlir/lib/Quantum/Transforms/remove_chained_self_inverse.cpp

```diff
@@ -45,9 +47,25 @@ struct RemoveChainedSelfInversePass
 
     void runOnOperation() final
     {
+        // Run the pass on a qnode whose name is "FuncNameOpt" from command line
+        // This option should only be used internally in frontend/catalyst/compiler.py
+        // when registering the pipelines and specifying what functions (qnodes) to run on from
+        // the pipeline table.
+        // In other words, compiler.py always uses this pass as a "FunctionPass", not a
+        // "ModulePass". If used from command line (./quantum-opt) directly and no value is
+        // supplied to FuncNameOpt, run the pass on all functions in the module
         LLVM_DEBUG(dbgs() << "remove chained self inverse pass"
                           << "\n");
 
+        Operation *op = getOperation();
+        if (isa<func::FuncOp>(op)) {
+            StringRef funcName = cast<func::FuncOp>(op).getSymName();
+
+            if (funcName != FuncNameOpt) {
+                return; // not the function to run the pass on, so do nothing and exit
+            }
+        }
```

Comment:
You can ensure that getOperation returns a func::FuncOp. https://mlir.llvm.org/docs/PassManagement/#filtered-operation-pass

Reply:
done